File: Backend/DevTrack.Api/Entities/Application.cs
namespace DevTrack.Api.Entities;

public static class ApplicationStatuses
{
    public const string Saved     = "Saved";
    public const string Applied   = "Applied";
    public const string Screening = "Screening";
    public const string Interview = "Interview";
    public const string Offer     = "Offer";
    public const string Rejected  = "Rejected";
    public const string Withdrawn = "Withdrawn";

    public static readonly IReadOnlySet<string> All = new HashSet<string>
    {
        Saved, Applied, Screening, Interview, Offer, Rejected, Withdrawn
    };
}

public class Application
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public string JobDescriptionRaw { get; set; } = string.Empty;

    public string? CompanyName { get; set; }
    public string? JobTitle { get; set; }
    public ParsedApplicationData? ParsedData { get; set; }

    public int? MatchScore { get; set; }
    public string? MatchExplanation { get; set; }

    public string Status { get; set; } = ApplicationStatuses.Saved;
    public DateTime? AppliedAt { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<ApplicationDocument> Documents { get; set; } = [];
    public ICollection<ApplicationStatusHistory> StatusHistory { get; set; } = [];
    public ICollection<Reminder> Reminders { get; set; } = [];
}

File: Frontend/src/types/application.ts
export type ApplicationStatus =
  | 'Saved'
  | 'Applied'
  | 'Screening'
  | 'Interview'
  | 'Offer'
  | 'Rejected'
  | 'Withdrawn'

export interface ParsedApplicationData {
  stack: string[] | null
  seniorityLevel: string | null
  remotePolicy: string | null
  salaryMin: number | null
  salaryMax: number | null
  salaryCurrency: string | null
  location: string | null
  postedAt: string | null
}

export interface DocumentDto {
  id: string
  type: string
  content: string
  createdAt: string
  updatedAt: string
}

export interface StatusHistoryDto {
  id: string
  fromStatus: string | null
  toStatus: string
  changedAt: string
  note: string | null
}

export interface ReminderDto {
  id: string
  type: string
  scheduledFor: string
  sentAt: string | null
  status: string
  createdAt: string
}

export interface Application {
  id: string
  companyName: string | null
  jobTitle: string | null
  parsedData: ParsedApplicationData | null
  matchScore: number | null
  matchExplanation: string | null
  status: ApplicationStatus
  appliedAt: string | null
  createdAt: string
  jobDescriptionRaw: string
}

export interface ApplicationDetail extends Application {
  documents: DocumentDto[]
  statusHistory: StatusHistoryDto[]
  reminders: ReminderDto[]
}

export interface CreateApplicationRequest {
  jobDescriptionRaw: string
}
